src/context.jl

```diff
@@ -70,6 +70,7 @@ Cassette.overdub(::Ctx, ::typeof(Core.kwfunc), f) = return Core.kwfunc(f)
 @inline Cassette.overdub(::Ctx, ::typeof(Base.isprimitivetype), t) = return Base.isprimitivetype(t)
 @inline Cassette.overdub(::Ctx, ::typeof(Base.isbitstype), t) = return Base.isbitstype(t)
 @inline Cassette.overdub(::Ctx, ::typeof(Base.isbits), x) = return Base.isbits(x)
+@inline Cassette.overdub(::Ctx, ::typeof(StaticArrays.Size), x::Type{<:AbstractArray{<:Any, N}}) where {N} = return StaticArrays.Size(x)
```

Comment:
```suggestion
# https://github.com/JuliaArrays/StaticArrays.jl/blob/e0d5f09f033ef2dea9289aba18ac0cf29c62037f/src/traits.jl#L65-L67
@inline Cassette.overdub(::Ctx, ::typeof(StaticArrays.Size), args...)=  return StaticArrays.Size(args...)
```

Since there are multiple `Size` methods

Reply:
I tried that last night and got this from the new troubled kernel:
```
ERROR: Error while loading expression starting at /home/jekozdon/codes/CLIMA/src/DGmethods/profiling/test.jl:11
caused by [exception 1]
GPU compilation of #12(CuDeviceArray{Float64,2,CUDAnative.AS.Global}) failed
KernelError: kernel returns a value of type `Union{}`

Make sure your kernel function ends in `return`, `return nothing` or `nothing`.
If the returned value is of type `Union{}`, your Julia code probably throws an exception.
Inspect the code with `@device_code_warntype` for more details.

Stacktrace:
 [1] check_method(::CUDAnative.CompilerJob) at /home/jekozdon/.julia/packages/CUDAnative/wU0tS/src/compiler/validation.jl:16
 [2] #codegen#116(::Bool, ::Bool, ::Bool, ::Bool, ::Bool, ::typeof(CUDAnative.codegen), ::Symbol, ::CUDAnative.CompilerJob) at /home/jekozdon/.julia/packages/TimerOutputs/7zSea/src/TimerOutput.jl:216
 [3] #codegen at /home/jekozdon/.julia/packages/CUDAnative/wU0tS/src/compiler/driver.jl:0 [inlined]
 [4] #compile#115(::Bool, ::Bool, ::Bool, ::Bool, ::Bool, ::typeof(CUDAnative.compile), ::Symbol, ::CUDAnative.CompilerJob) at /home/jekozdon/.julia/packages/CUDAnative/wU0tS/src/compiler/driver.jl:47
 [5] #compile#114 at ./none:0 [inlined]
 [6] compile at /home/jekozdon/.julia/packages/CUDAnative/wU0tS/src/compiler/driver.jl:28 [inlined] (repeats 2 times)
 [7] macro expansion at /home/jekozdon/.julia/packages/CUDAnative/wU0tS/src/execution.jl:378 [inlined]
 [8] #cufunction#156(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(cufunction), ::getfield(GPUifyLoops, Symbol("##12#13")){typeof(kernel_similar_MArray!)}, ::Type{Tuple{CuDeviceArray{Float64,2,CUDAnative.AS.Global}}}) at /home/jekozdon/.julia/packages/CUDAnative/wU0tS/src/execution.jl:347
 [9] cufunction(::Function, ::Type) at /home/jekozdon/.julia/packages/CUDAnative/wU0tS/src/execution.jl:347
 [10] macro expansion at /home/jekozdon/codes/julia_dev/GPUifyLoops/src/GPUifyLoops.jl:113 [inlined]
 [11] macro expansion at ./gcutils.jl:87 [inlined]
 [12] #launch#46(::Base.Iterators.Pairs{Symbol,Tuple{Int64,Int64},Tuple{Symbol},NamedTuple{(:threads,),Tuple{Tuple{Int64,Int64}}}}, ::typeof(GPUifyLoops.launch), ::CUDA, ::typeof(kernel_similar_MArray!), ::CuArray{Float64,2}) at /home/jekozdon/codes/julia_dev/GPUifyLoops/src/GPUifyLoops.jl:110
 [13] (::getfield(GPUifyLoops, Symbol("#kw##launch")))(::NamedTuple{(:threads,),Tuple{Tuple{Int64,Int64}}}, ::typeof(GPUifyLoops.launch), ::CUDA, ::typeof(kernel_similar_MArray!), ::CuArray{Float64,2}) at ./none:0
 [14] top-level scope at /home/jekozdon/codes/julia_dev/GPUifyLoops/src/GPUifyLoops.jl:54
 [15] top-level scope at /home/jekozdon/codes/CLIMA/src/DGmethods/profiling/test.jl:17
 [16] include at ./boot.jl:328 [inlined]
 [17] include_relative(::Module, ::String) at ./loading.jl:1094
 [18] include(::Module, ::String) at ./Base.jl:31
 [19] include(::String) at ./client.jl:431
 [20] top-level scope at REPL[1]:1
```

Reply:
Okay that is worthwhile investigating. Cthulhu should be able to step into the GPU code.